arrows/ocv/resection_camera.cxx

```diff
@@ -0,0 +1,198 @@
+namespace arrows {
+namespace ocv
+{
+
+/// private implementation
+struct resection_camera::priv
+{
+  priv()
+    : confidence_threshold(0.99)
+    , max_iterations(10000)
+    , m_logger( vital::get_logger( "arrows.ocv.resection_camera" ))
+  {
+  }
+
+  double confidence_threshold;
+  int max_iterations;
+
+  vital::logger_handle_t m_logger;
+};
+
+resection_camera::resection_camera()
+: d_(new priv)
+{
+}
+
+resection_camera::~resection_camera()
+{
+}
+
+/// Get this algorithm's \link vital::config_block configuration block \endlink
+vital::config_block_sptr
+resection_camera::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config =
+      vital::algo::resection_camera::get_configuration();
+  config->set_value("confidence_threshold", d_->confidence_threshold,
+    "Confidence that estimated matrix is correct, range (0.0, 1.0]");
+  config->set_value("max_iterations", d_->max_iterations,
+    "maximum number of iterations to run PnP [1, INT_MAX]");
+  return config;
+}
+
+/// Set this algorithm's properties via a config block
+void
+resection_camera::set_configuration(vital::config_block_sptr config)
+{
+  d_->confidence_threshold =
+    config->get_value<double>("confidence_threshold", d_->confidence_threshold);
+  d_->max_iterations =
+    config->get_value<int>("max_iterations", d_->max_iterations);
+}
+
+/// Check that the algorithm's configuration vital::config_block is valid
+bool
+resection_camera::check_configuration(vital::config_block_sptr config) const
+{
+  bool good_conf = true;
+  double confidence_threshold =
+    config->get_value<double>("confidence_threshold", d_->confidence_threshold);
+
+  if( confidence_threshold <= 0.0 || confidence_threshold > 1.0 )
+  {
+    LOG_ERROR(d_->m_logger, "confidence_threshold parameter is "
+                            << confidence_threshold
+                            << ", needs to be in (0.0, 1.0].");
+    good_conf = false;
+  }
+
+  int max_iterations = config->get_value<int>("max_iterations", d_->max_iterations);
+
+  if (max_iterations < 1)
+  {
+    LOG_ERROR(d_->m_logger, "max iterations is " << max_iterations
+      << ", needs to be greater than zero.");
+    good_conf = false;
+  }
+  return good_conf;
+}
+
+kwiver::vital::camera_perspective_sptr
+resection_camera::resection(
+	vector<kwiver::vital::vector_2d> const & pts2d,
+    vector<kwiver::vital::vector_3d> const & pts3d,
+    vector<bool>& inliers,
+    kwiver::vital::camera_intrinsics_sptr cal
+) const
+{
+  if (cal==nullptr)
+  {
+    LOG_ERROR(d_->m_logger, "camera calibration guess should not be null in this implementation");
+    return vital::camera_perspective_sptr();
+  }
+  const unsigned
+    minCnt = 3,
+    pts2cnt = pts2d.size(),
+    pts3cnt = pts3d.size();
+  if (pts2cnt < minCnt || pts3cnt < minCnt)
+  {
+    LOG_ERROR(d_->m_logger, "not enough points to resection camera");
+    return vital::camera_perspective_sptr();
+  }
+  if (pts2cnt != pts3cnt)
```

Comment:
OK, unsure why one-liners deserve scoping, but will follow the style for consistency.

Reply:
It's a little more verbose, but it can remove ambiguities and make code more readable.  Here is some discussion on the topic:
https://stackoverflow.com/questions/12193170/whats-the-purpose-of-using-braces-i-e-for-a-single-line-if-or-loop